src/data_hub/plotting.py

```diff
@@ -38,22 +43,58 @@ def scatter(
 
     """
 
+    # constants for this plot
     figure_width = 500
     select_width = int(figure_width * 0.3)
-    js_code = _read_js_code()
+    if js_code is None:
+        js_code = _read_js_code()
+    default_label = "None"
+    default_color = "blue"
 
     cols = df.attrs["metadata"]["cols"]
     numeric_cols = [col["name"] for col in cols if col["type"] in ["float", "integer"]]
 
-    label_data = None
-
-    select_cols = [
-        col for col in df.attrs["metadata"]["cols"] if col["type"] == "select"
-    ]
-    if len(select_cols) > 0:
-        select_col_names = [col["name"] for col in select_cols]
-        select_col_options = [col["configSelect"]["options"] for col in select_cols]
-        label_data = dict(zip(select_col_names, select_col_options))
+    # make placeholder lists for legends and colors
+    legend_labels = [default_label for _ in range(len(df))]
+    colors = [default_color for _ in range(len(df))]
+
+    if label_column is not None:
+        select_cols = [
+            col for col in df.attrs["metadata"]["cols"] if col["type"] == "select"
+        ]
+        if len(select_cols) == 0:
+            raise DeepOriginException("No select columns detected in the dataframe.")
+        select_cols = [col["name"] for col in select_cols]
+        if label_column not in select_cols:
+            raise DeepOriginException(
+                f"Column '{label_column}' is not a select column in the dataframe."
+            )
+
+        labels = df[label_column].cat.categories.tolist()
+
+        if len(labels) == 0:
+            raise DeepOriginException(
+                f"The column {label_column} has no options configured, and therefore cannot be used to label data."
+            )
+
+        if len(labels) > 10:
+            raise DeepOriginException(
+                f"The column {label_column} has too many ({len(labels)}) options configured, and therefore cannot be used to label data."
+            )
+
+        # make color map
+        cat10_colors = Category10
+        cat10_colors[1] = cat10_colors[3][:1]
+        cat10_colors[2] = cat10_colors[3][:2]
+        cat10_colors = cat10_colors[len(labels)]
+        color_map = {label: color for label, color in zip(labels, cat10_colors)}
+        color_map[default_label] = default_color
```

Comment:
this inscrutable code makes a colormap, which works for anything from 1-10 colors